src/index.js

```diff
@@ -465,14 +465,20 @@ function createConfig(options, entry, format, writeMeta) {
 	if (options.external === 'none') {
 		// bundle everything (external=[])
 	} else if (options.external) {
-		external = external.concat(peerDeps).concat(options.external.split(','));
+		external = external.concat(peerDeps).concat(
+			// CLI --external supports regular expressions:
+			options.external.split(',').map(str => new RegExp(str)),
+		);
 	} else {
 		external = external
 			.concat(peerDeps)
 			.concat(Object.keys(pkg.dependencies || {}));
 	}
 
 	let globals = external.reduce((globals, name) => {
+		// Use raw/escaped value for CLI-provided RegExp externals:
```

Comment:
Do you mean "Use raw/unescaped value"? (Since RegExp.source will not return a "fully" escaped regex https://jsfiddle.net/h6g1s39c/)

Reply:
Yes, sorry - the comment was confusing/wrong. I think we just want the raw value here?  TBH in the case where an actual regex is provided (`--external .*`) we can't build a useful name anyway - it'll fail the check on L481.